core/src/main/java/org/elasticsearch/common/io/stream/StreamOutput.java

```diff
@@ -413,6 +414,34 @@ public void writeMap(@Nullable Map<String, Object> map) throws IOException {
     }
 
     /**
+     * write map to stream with consistent order
+     * to make sure every map generated bytes order are same
+     */
+    public void writeMapWithConsistentOrder(@Nullable Map<String, Object> map) throws IOException {
+        if (map == null) {
+            writeByte((byte) -1);
+            return;
+        }
+        if (map instanceof LinkedHashMap) {
+            this.writeByte((byte) 9);
```

Comment:
I think sorting a linked hash map is a bit weird to do though. Maybe just `assert false == map instanceof LinkedHashMap; writeByte((byte) 10);`